src/main/kotlin/io/github/gradlenexus/publishplugin/CloseNexusStagingRepository.kt

```diff
@@ -16,21 +16,13 @@
 
 package io.github.gradlenexus.publishplugin
 
-import io.github.gradlenexus.publishplugin.internal.InvalidatingStagingRepositoryDescriptorRegistry
 import io.github.gradlenexus.publishplugin.internal.StagingRepositoryTransitioner
-import org.gradle.api.model.ObjectFactory
-import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.options.Option
-import javax.inject.Inject
 
-abstract class CloseNexusStagingRepository @Inject constructor(
-    objects: ObjectFactory,
-    repository: NexusRepository,
-    registry: Provider<InvalidatingStagingRepositoryDescriptorRegistry>
-) : AbstractTransitionNexusStagingRepositoryTask(objects, repository, registry) {
+abstract class CloseNexusStagingRepository : AbstractTransitionNexusStagingRepositoryTask() {
```

Comment:
I noticed that only abstract tasks are named "...Task", the code might be easier to read if all of them have the suffix, similar to how ...Plugin and ...Extension have suffixes. This would be a bigger breaking change, one that might actually affect users (tasks.withType). Thoughts @szpak?

Reply:
Yeah, definitely separate PR, in IJ Gradle Plugin they did this which is really cool:
https://github.com/JetBrains/gradle-intellij-plugin/blob/89ae190a30b5392ddc36bb02cc4042458c688142/src/main/kotlin/org/jetbrains/intellij/tasks/InstrumentCodeTask.kt#L27C1-L28

But it only works for Kotlin, Java and Groovy users will still get a "missing class" compile error. And any transitive projects will also fail as v1 and v2 will have different classes.

If we do the rename and reintroduce extending classes as you're showing, we would still need to register the old names in the TaskContainer so that `tasks.withType` keeps working. That said, if you want a deprecation cycle, that's the only thing we can do without seriously breaking all users of `withType` (and whatever other use case I'm not thinking of).

Keep in mind that deprecation warnings are not always shown for build.gradle files, so it might be hard for some users to notice.

Reply:
> Keep in mind that deprecation warnings are not always shown for build.gradle files, so it might be hard for some users to notice.

You are right, we would need to do some warning logging before delegation or use some internal Gradle magic for presenting deprecation warnings. To limit (that) plugin maintenance cost, I'm open to just rename the classes - 2.0.0 might provide some surprises :-).

Reply:
Alternatively, we can just leave it as is :)

Anyway, let's see how it would look like in a separate PR and continue discussion there.

Reply:
Opened https://github.com/TWiStErRob/nexus-publish-gradle-plugin/pull/6 for now until this gets merged.